### PR TITLE
bpo-1635741: Port time module to multiphase initialization (PEP 489)

### PR DESCRIPTION
**Pull request by phsilva**
https://bugs.python.org/issue1635741

### CODE REVIEW

File: Modules/timemodule.c [hunk 1]
```diff
@@ -1736,27 +1736,9 @@ if it is 1, the time is given in the DST time zone;\n\
...
-    NULL,
-    NULL,
-    NULL,
-    NULL
-};
-
-PyMODINIT_FUNC
-PyInit_time(void)
+static int
+time_exec(PyObject *m)
```
Comment (shihai1991) [inline Modules/timemodule.c]: Use `mod` or `module` would be more exact than `m` ;)

**Comment by shihai1991** [comment]
cc @vstinner @corona10 

ps: I don't know what happend with the gate of macOS(Looks like it have been broken several days.)

**Comment by corona10** [comment]
@shihai1991 
This is the known issue :)
https://bugs.python.org/issue40014

**Comment by shihai1991** [comment]
> @shihai1991
> This is the known issue :)
> https://bugs.python.org/issue40014

copy that, thanks ;0

File: Modules/timemodule.c (hunk 1 above)
Comment (vstinner) [reply Modules/timemodule.c]: Yeah, please this PR as an opportunity to replace m with module ;-)

Comment (shihai1991) [reply Modules/timemodule.c]: I have an impractical idea: Making an review robot to give some suggestions of normative problem automatically ;)

File: Modules/timemodule.c [hunk 2]
```diff
@@ -1840,11 +1822,34 @@ PyInit_time(void)
         utc_string = tm.tm_zone;
 #endif
 
-    return m;
+    return 0;
 
 error:
-    Py_DECREF(m);
-    return NULL;
+    Py_DECREF(module);
```
Comment (vstinner) [inline Modules/timemodule.c]: DECREF here is wrong. Replace "goto error" with "return -1" and remove the error label.

cc @corona10

Comment (phsilva) [reply Modules/timemodule.c]: I started not wanting to refactor code but looks like a good chance for more cleaning. Thanks, already changed on latest push.

**Review by vstinner — APPROVED** [review]
Ok, this time it LGTM. Thanks ;-)

**Pull request merged**